### ISSUE TITLE
ReferenceError: window is not defined at UniversalStorage.setUniversalItem after switching from 5.0.9 to 5.0.10

### ISSUE THREAD

**Issue by asmajlovicmars** [aws-amplify/amplify-js#10869]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Not applicable

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line
JS framework: SvelteKit

  System:
    OS: macOS 12.6.2
    CPU: (10) arm64 Apple M1 Pro
    Memory: 78.98 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Firefox: 108.0.2
    Safari: 16.2
  npmPackages:
    @aws-sdk/client-cloudwatch-logs: ^3.250.0 => 3.250.0
    @aws-sdk/client-cognito-identity: ^3.245.0 => 3.245.0
    @aws-sdk/client-s3: ^3.245.0 => 3.245.0
    @aws-sdk/client-ssm: ^3.245.0 => 3.245.0
    @aws-sdk/credential-provider-cognito-identity: ^3.245.0 => 3.245.0
    @aws-sdk/middleware-endpoint: ^3.226.0 => 3.226.0
    @aws-sdk/s3-request-presigner: ^3.245.0 => 3.245.0
    @aws-sdk/util-endpoints: ^3.245.0 => 3.245.0
    @contentful/rich-text-html-renderer: ^16.0.2 => 16.0.2
    @contentful/rich-text-types: ^16.0.2 => 16.0.2
    @cypress/code-coverage: ^3.10.0 => 3.10.0
    @floating-ui/dom: ^1.1.0 => 1.1.0
    @fontsource/fira-mono: ^4.5.10 => 4.5.10
    @googlemaps/js-api-loader: ^1.15.1 => 1.15.1
    @iconify/json: ^2.2.6 => 2.2.6
    @lukeed/uuid: ^2.0.0 => 2.0.0
    @playwright/experimental-ct-svelte: ^1.29.2 => 1.29.2
    @playwright/test: ^1.29.2 => 1.29.2
    @popperjs/core: ^2.11.6 => 2.11.6
    @portabletext/svelte: ^1.0.1 => 1.0.1
    @sanity/client: ^3.4.1 => 3.4.1
    @sanity/image-url: ^1.0.1 => 1.0.1
    @sentry/browser: ^7.30.0 => 7.30.0
    @sentry/node: ^7.30.0 => 7.30.0
    @sentry/tracing: ^7.30.0 => 7.30.0
    @sveltejs/adapter-node: ^1.1.1 => 1.1.1
    @sveltejs/kit: ^1.1.0 => 1.1.0
    @sveltejs/pancake: ^0.0.18 => 0.0.18
    @tailwindcss/typography: ^0.5.9 => 0.5.9
    @types/cookie: ^0.5.1 => 0.5.1
    @types/crypto-js: ^4.1.1 => 4.1.1
    @types/jsforce: ^1.11.0 => 1.11.0
    @types/ramda: ^0.28.20 => 0.28.20
    @typescript-eslint/eslint-plugin: ^5.48.1 => 5.48.1
    @typescript-eslint/parser: ^5.48.1 => 5.48.1
    @zerodevx/svelte-toast: ^0.8.2 => 0.8.2
    autoprefixer: ^10.4.13 => 10.4.13
    aws-amplify: 5.0.10 => 5.0.10
    aws-sdk: ^2.1295.0 => 2.1295.0
    classnames: ^2.3.2 => 2.3.2
    contentful: ^9.2.15 => 9.2.15
    contentful-sitemap: ^1.0.20 => 1.0.20
    cookie: ^0.5.0 => 0.5.0
    crocks: ^0.12.4 => 0.12.4
    crypto-js: ^4.1.1 => 4.1.1
    cypress: ^12.3.0 => 12.3.0
    daisyui: ^2.46.1 => 2.46.1
    dayjs: ^1.11.7 => 1.11.7
    dotenv: ^16.0.3 => 16.0.3
    env-cmd: ^10.1.0 => 10.1.0
    eslint: ^8.31.0 => 8.31.0
    eslint-config-prettier: ^8.6.0 => 8.6.0
    eslint-plugin-svelte3: ^4.0.0 => 4.0.0
    flowbite-svelte: 0.29.7 => 0.29.7
    graphql: ^16.6.0 => 16.6.0
    graphql-request: ^5.1.0 => 5.1.0
    jsforce: ^1.11.0 => 1.11.0
    jwt-decode: ^3.1.2 => 3.1.2
    mdsvex: ^0.10.6 => 0.10.6
    mocha: ^10.2.0 => 10.2.0
    mochawesome: ^7.1.3 => 7.1.3
    nyc: ^15.1.0 => 15.1.0
    postcss: ^8.4.21 => 8.4.21
    postcss-load-config: ^4.0.1 => 4.0.1
    prettier: ^2.8.2 => 2.8.2
    prettier-plugin-svelte: ^2.9.0 => 2.9.0
    ramda: ^0.28.0 => 0.28.0
    rollup-plugin-istanbul: ^4.0.0 => 4.0.0
    svelte: ^3.55.1 => 3.55.1
    svelte-check: ^2.10.3 => 2.10.3
    svelte-heros: ^2.4.3 => 2.4.3
    svelte-imask: ^1.2.0 => 1.2.0
    svelte-preprocess: ^5.0.0 => 5.0.0
    svelte-select: ^5.1.1 => 5.1.1
    tailwindcss: ^3.2.4 => 3.2.4
    tslib: ^2.4.1 => 2.4.1
    typescript: ^4.9.4 => 4.9.4
    unplugin-icons: ^0.14.15 => 0.14.15
    uvu: ^0.5.6 => 0.5.6
    vite: ^4.0.4 => 4.0.4
    zod: ^3.20.2 => 3.20.2
  npmGlobalPackages:
    @winglang/wing: 0.3.25
    aws-cdk: 2.55.1
    corepack: 0.14.2
    npm: 8.19.2


```

</details>

### Describe the bug

After upgrading from aws-amplify from 5.0.9 to 5.0.10 Auth.signIn stopped working.

```
Amplify.configure({ ...awsmobile, ssr: true });
const { Auth } = withSSRContext({ req });
let cogUser = await Auth.signIn(email, password);
```

Error: 

```
ReferenceError: window is not defined at UniversalStorage.setUniversalItem (./app/node_modules/.pnpm/@aws-amplify+core@5.0.10/node_modules/@aws-amplify/core/lib/UniversalStorage/index.js:87:21)
    at UniversalStorage.setItem (./app/node_modules/.pnpm/@aws-amplify+core@5.0.10/node_modules/@aws-amplify/core/lib/UniversalStorage/index.js:73:18)
    at CognitoUser.cacheTokens (./app/node_modules/.pnpm/amazon-cognito-identity-js@6.1.2/node_modules/amazon-cognito-identity-js/lib/CognitoUser.js:1217:18)
    at CognitoUser.authenticateUserInternal (./app/node_modules/.pnpm/amazon-cognito-identity-js@6.1.2/node_modules/amazon-cognito-identity-js/lib/CognitoUser.js:411:10)
    at ./app/node_modules/.pnpm/amazon-cognito-identity-js@6.1.2/node_modules/amazon-cognito-identity-js/lib/CognitoUser.js:298:27
    at ./app/node_modules/.pnpm/amazon-cognito-identity-js@6.1.2/node_modules/amazon-cognito-identity-js/lib/CognitoUser.js:281:22
    at ./app/node_modules/.pnpm/amazon-cognito-identity-js@6.1.2/node_modules/amazon-cognito-identity-js/lib/Client.js:123:31
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

### Expected behavior

Expect it to work as 5.0.9 works. 

### Reproduction steps

Just use aws-amplify signIn on the server side. 

```


```

### Code Snippet

```javascript
// Put your code below this line.

// invoke SSR Auth.signIn(x,y)
Amplify.configure({ ...awsmobile, ssr: true });
const { Auth } = withSSRContext({ req });
let cogUser = await Auth.signIn(email, password);



// aws-amplify code that causes an error:
// location: ./node_modules/.pnpm/@aws-amplify+core@5.0.10/node_modules/@aws-amplify/core/lib/UniversalStorage/index.js
    UniversalStorage.prototype.setUniversalItem = function (key, value, options) {
        if (options === void 0) { options = {}; }
        this.cookies.set(key, value, tslib_1.__assign(tslib_1.__assign({}, options), { path: '/', 
            // `httpOnly` cannot be set via JavaScript: https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#JavaScript_access_using_Document.cookie
            sameSite: true, 
            // Allow unsecure requests to http://localhost:3000/ when in development.
            secure: window.location.hostname === 'localhost' ? false : true }));    // <------ issue 
            // a.s. above line of code is not checking if isBrowser, and assumes that window is available! 
    };


```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by AllanZhengYP** [comment]
Hi @asmajlovicmars,

Thank you for reporting this. It's indeed a bug introduced by mentioned PR. I already put up a fix for it. 

I'd like to understand more of your use case. Do you call the `Auth.signIn` on server side? How do you transfer the credentials(username, password etc.) securely to backend?

**Comment by asmajlovicmars** [comment]
Hi @AllanZhengYP,
Thank you for fixing this 🙏 

Yes, we call Auth.signIn on the server side, to avoid exposing any of the secrets. We use the amazing SvelteKit, which now has PUBLIC and PRIVATE env vars, so all secrets are picked up from the AWS Parameter Store, and then only served on the server side as PRIVATE, which is impossible to import on the FE side, even by mistake. 

In terms of the sign-in email/password: CryptoAES FE to BE.

**Comment by AllanZhengYP** [comment]
Thank you @asmajlovicmars for sharing your config! Using the server-side environment veriables is indeed a valid use case. 

The fix has been released in [5.0.11](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%405.0.11), please give a try.

I will close the issue for now. Please feel free to create new issues if you have more questions!
